src/Parser/Monk/Mistweaver/Modules/Talents/RisingMist.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { STATISTIC_ORDER } from 'Main/StatisticBox';
+import ExpandableStatisticBox from 'Main/ExpandableStatisticBox';
+import { formatNumber } from 'common/format';
+import SpellIcon from 'common/SpellIcon';
+
+import SPELLS from 'common/SPELLS';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import HotTracker from '../Core/HotTracker';
+
+const debug = false;
+
+const RISING_MIST_EXTENSION = 2000;
+
+class RisingMist extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    hotTracker: HotTracker,
+  };
+
+  risingMistCount = 0;
+  risingMists = [];
+
+  efsExtended = 0; // tracks how many flourishes extended Wild Growth
+
+  remCount = 0;
+  efCount = 0;
+  evmCount = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.RISING_MIST.id);
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (SPELLS.RISING_SUN_KICK.id !== spellId) {
+      return;
+    }
+
+    this.risingMistCount += 1;
+    debug && console.log(`risingMist cast #: ${this.risingMistCount}`);
+
+    const newRisingMist = {
+      name: `risingMist #${this.risingMistCount}`,
+      healing: 0,
+      procs: 0,
+      duration: 0,
+    };
+    this.risingMists.push(newRisingMist);
+
+    let foundEf = false;
+
+    Object.keys(this.hotTracker.hots).forEach(playerId => {
```

Comment:
Oof, okay - this could be improved a good bit I think. Any time you find yourself assigning to an object from a forEach, you should probably consider reduce as a good way to make that code a bit more readable. To make this even more verbose, we can use `Object.entries` to avoid keying the object and using that to access the associated values. 

Here's some quick code to demonstrate a similar problem:
```js
const hotTest = {
  reglitch: ['3', '4'],
  oratio: ['2'],
  zero: ['1', '2', '4'],
};

// Should be obvious
const strToNumber = str => Number(str) || undefined;

// Log occurences of a variable to an object
const logToObject = (arr, obj = {}) =>
  arr.reduce((acc, curr) => {
    if (acc[curr]) {
      acc[curr] += 1;
      return acc;
    }

    acc[curr] = 1;
    return acc;
  }, obj);

// Build a k:v pair of how often spells are used given an array of spellIds assigned to playerIds
// ​​​​​Output here is: { '1': 1, '2': 2, '3': 1, '4': 2 }​​​​​
const spellUsageLog = Object.entries(hotTest).reduce((acc, curr) => {
  const [player, spellIds] = curr;
  const spellsUsed = spellIds.map(strToNumber);

  // Put code to add extension to hot tracker here lol
  const logSpells = logToObject(spellsUsed, acc);

  return logSpells;
}, {});
```

You'll need to filter for the set of spells you want to account for but that should be trivial.

Reply:
I think he was copying my Flourish code... but yeah ok it's not the prettiest thing I've written.